File: Python/insertion_sort.py
import sys

def insertionSort1(n, arr):
    
    for i in range(0,len(arr)):
        key=arr[i]
        j=i-1

        while (key < arr[j]) and (j >= 0):
            arr[j+1] = arr[j]
            print(' '.join(map(str, arr)))
            j -= 1
        
    arr[j+1] = key
    print(' '.join(map(str, arr)))
    
if __name__ == "__main__":
    n = int(input().strip())
    arr = list(map(int, input().strip().split(' ')))
    insertionSort1(n, arr)

File: Go/insertion_sort.go
package main

import "fmt"

func main() {

  
    len := 0
    fmt.Print("Enter the number of elements in array : ")
    fmt.Scanln(&len)
    input := make([]int, len)
    for i := 0; i < len; i++ {
        fmt.Print("Enter a number : ")
        fmt.Scanf("%d", &input[i])
    }
    insertionSort(input)
}

func insertionSort(arr []int) {
    len := len(arr)
    for i := 1; i < len; i++ {
        for j := 0; j < i; j++ {
            if arr[j] > arr[i] {
                arr[j], arr[i] = arr[i], arr[j]
            }
        }
    }
  fmt.Print("Sorted Array : ")
    for _, val := range arr {
        fmt.Print(val," ")
    }
}
